### ISSUE TITLE
Run administrative tasks in their own scheduling group

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#4970]
Operations like `nodetool cleanup` can be CPU-intensive. They should be isolated from other workloads.

It was observed that `nodetool cleanup` starves other processes of CPU, which leads to unavailability and OOM.

**Comment by glommer** [comment]
Tomek I believe you observed this in an old version that doesnt have a CPU scheduler. Are you sure this happens in newer post 2.3 versions ?

**Comment by tgrabiec** [comment]
I am not sure what exactly would happen on more recent versions, but looking at the code, I don't see RESTful API calls being executed in a separate scheduling group, so they would run (and block) the main scheduling group.

**Comment by avikivity** [comment]
```c++
            with_scheduling_group(maintenance_scheduling_group, [&] {
                return ctx.http_server.listen(ipv4_addr{ip, api_port});
            }).get();
```

**Comment by avikivity** [comment]
This was added in 3.0.

**Comment by tgrabiec** [comment]
Fixed in e7ae4beef00fbd8b17406ce01281ec3b3d606117
